### DISCUSSION TITLE
npm publish and tag releases

### DISCUSSION THREAD

**Discussion by theoludwig** [npm/feedback#194] [CLI] [answered]
Hey 👋,

To automate the deployment of my npm packages, I use GitHub Actions that triggers everytime we create a release/tag on GitHub, this action use `npm publish --access public` command.
I also wanted to automate the deployment of the [standard](https://www.npmjs.com/package/standard) package, so I opened a PR to implement this : https://github.com/standard/standard/pull/1634

But currenly we are only releasing with latest tag, how does it it work if we want to release on a beta tag for example ?

I thought you only needed to change the version number in `package.json` so it can release on another tag, but as it is in the [doc](https://docs.npmjs.com/cli/v7/commands/npm-publish), we also need to run `npm publish --tag=beta` for example, it doesn't automatically, release on the tag specified in the version number inside `package.json`.

In my opinion, it should be the case, I ran `npm publish --dry-run` and here is what I've got :
![2021-02-04_07-16](https://user-images.githubusercontent.com/25207499/106852865-114a5880-66b9-11eb-814e-dfa101d8c139.png)

I don't know what means `standard@16.0.3-beta1`, is it still released under latest tag with this version number or effectively release with beta tag ?
(I only updated `package.json` version number to `16.0.3-beta1`, nothing else)

Thanks for your answer and your help! :smile:

**Comment by arcanis** [comment]
> I don't know what means `standard@16.0.3-beta1`, is it still released under latest tag with this version number or effectively release with beta tag ?

Versions numbers have no direct connection with dist-tags, so you can have a `beta` tag with stable versions, and `latest` with beta qualifiers.

As a quick tip, avoid using `beta1` as beta qualifier, prefer using `beta.1`. If you don't you'll hit issues if you ever need to release a bunch of betas (because as per semver, `beta10 < beta2` due to lexicographic sort).

**Reply by theoludwig** [reply]
Oh right, also thanks for your tip, we'll use `standard@16.0.3-beta.1` instead of the `beta1` one.

> Versions numbers have no direct connection with dist-tags

But shouldn't be the case ? In my opinion it feels weird to have a version release to latest tag with the number version as beta, don't you think ?

What are the use cases to have the version `16.0.3-beta.1` released under latest tag ?
Souldn't be by default release to `beta` tag ? It certainly need a little bit more code CLI part since we need to parse the version number and to see if there is a tag name. Is `yarn` working same way as npm to release packages ? 

I don't see any use case to this :
> so you can have a `beta` tag with stable versions, and `latest` with beta qualifiers.

**Reply by arcanis** [reply]
> Is `yarn` working same way as npm to release packages ?

I think we do (we don't change the target tag based on the version), but if you want to open an issue on our repo it's probably something we can discuss for the next major.

**Reply by theoludwig** [reply]
Issue available there : https://github.com/yarnpkg/berry/issues/2444
Also it would be cool that JavaScipt package manager follow each other the features, so if we implement this on `yarn`, we should also consider it for `npm` and `pnpm` cli tools.

**Comment by MylesBorins** [comment, marked as answer]
At this time we still require the use of the `--tag` argument to set the specific dist-tag and don't assume anything based on the semver. To be honest I think it might be too fragile to do anything else. It might be worth kicking off an issue over at https://github.com/npm/rfcs if you want to propose a larger change to how this works with publish.

**Reply by theoludwig** [reply]
Thanks for your answer!
RFC open there : https://github.com/npm/rfcs/pull/317
